File: lib/rb/lib/thrift/types/annotation/naming/naming.rb
require 'thrift/types/annotation/naming/naming_types'

module Thrift
  module Types
    module Annotation
      module Naming
        class CanonicalNameExtractor
          class << self
            def extract(definition)
              definition.structured_annotations.select do |sa|
                sa.is_a? PreviouslyKnownAs
              end.map do |pka|
                "#{pka.namespace_ || definition.namespace}.#{pka.name || definition.name}"
              end
            end
          end

          ::Thrift.register_canonical_name_extractor self
        end
      end
    end
  end
end

File: lib/go/thrift/types/annotation/naming/naming.go
package naming

import "github.com/upfluence/thrift/lib/go/thrift/internal/reflection"

func init() {
	reflection.RegisterCanonicalNameExtractor(
		reflection.CanonicalNameExtractorFunc(func(ns string, def reflection.AnnotatedDefinition) []string {
			var res []string

			for _, sa := range def.StructuredAnnotations {
				if pka, ok := sa.(*PreviouslyKnownAs); ok {
					lns := pka.GetNamespace_()
					ln := pka.GetName()

					if pka.Namespace_ == nil {
						lns = ns
					}

					if ln == "" {
						ln = def.Name
					}

					res = append(res, lns+"."+ln)
				}
			}

			return res
		}),
	)
}
